corehq/ex-submodules/auditcare/utils/export.py

```diff
@@ -40,6 +40,14 @@ def _date_key(date):
     return ids
 
 
+def get_auditcare_docs_by_username(username):
+    event_ids = navigation_event_ids_by_user(username)
+    doc_list = []
+    for event in iter_docs(NavigationEventAudit.get_db(), event_ids):
+        doc_list.append(NavigationEventAudit.wrap(event))
```

Comment:
This will put all docs into memory, which could be a lot and cause OOM. Instead of this, you'll want to use `yield NavigationEventAudit.wrap(event)` and in the management command you'll use:

```
for doc in get_auditcare_docs_by_username(username):
    doc.user = new_username
    doc.save()
```

alternatively you can chunk your saves as well by using `bulk_save`, but probably not necessary

Reply:
Done here
https://github.com/dimagi/commcare-hq/pull/19972/commits/a7a61bfb7f28fc2b8efcdecfe1dd03a4e33ac3c2